pkg/alertmanager/alertmanager_ring.go

```diff
@@ -77,13 +80,15 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.HeartbeatPeriod, rfprefix+"heartbeat-period", 15*time.Second, "Period at which to heartbeat to the ring.")
 	f.DurationVar(&cfg.HeartbeatTimeout, rfprefix+"heartbeat-timeout", time.Minute, "The heartbeat timeout after which alertmanagers are considered unhealthy within the ring.")
 	f.IntVar(&cfg.ReplicationFactor, rfprefix+"replication-factor", 3, "The replication factor to use when sharding the alertmanager.")
+	f.BoolVar(&cfg.ZoneAwarenessEnabled, rfprefix+"zone-awareness-enabled", false, "True to enable zone-awareness and replicate alerts across different availability zones.")
```

Comment:
Cleaning up this is a little harder but not impossible. The `defaultPrefix` passed here is different (`alertmanagers/`) to the one hard-coded in [RegisterFlagsWithPrefix](https://github.com/cortexproject/cortex/blob/master/pkg/ring/ring.go#L143) (`collectors/`), so we could change that function to also accept `defaultPrefix`.

Reply:
Good suggestion @stevesg. However, I would keep it for a follow up PR. This PR is just adopting the same pattern we're already using in this function.